ydb/public/lib/json_value/ydb_json_value.cpp

```diff
@@ -745,11 +745,15 @@ namespace {
                     const TString& memberName = TypeParser.GetMemberName();
                     const auto it = jsonMap.find(memberName);
                     if (it == jsonMap.end()) {
-                        ThrowFatalError(TStringBuilder() << "No member \"" << memberName
-                            << "\" in the map in json string for YDB struct type");
+                        // missing JSON value is treated as NULL on input
+                        ValueBuilder.AddMember(memberName);
+                        TypeParser.OpenOptional();
```

Comment:
Better to check, that value kind is optional, because otherwise error will be confusing for user. For example, "Expected optional type because no member in the map in json string for YDB struct type"